frameworks/hdfs/src/main/dist/svc.yml

```diff
@@ -213,12 +220,31 @@ pods:
           cp -v zone-resolver.sh {{TASKCFG_ALL_HDFS_VERSION}}/etc/hadoop/zone-resolver.sh
           chmod u+x {{TASKCFG_ALL_HDFS_VERSION}}/etc/hadoop/zone-resolver.sh
           {{/NAME_NODE_PLACEMENT}}
-          ./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs namenode
+          {{#SECURITY_KERBEROS_ENABLED}}
+          chmod +x fix-keytab-bug.sh
+          ./fix-keytab-bug.sh secrets-hdfs.keytab
+          {{/SECURITY_KERBEROS_ENABLED}}
+          export LATEST_FSIMAGE_VERSION=-65
+          export CURRENT_FSIMAGE_VERSION=$(cat name-data/current/VERSION | grep "layoutVersion" | cut -d'=' -f2)
+          if [[ $CURRENT_FSIMAGE_VERSION -gt $LATEST_FSIMAGE_VERSION ]]; then
+            ./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs dfsadmin -rollingUpgrade prepare
+            ./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs dfsadmin -rollingUpgrade query
+            ./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs namenode -rollingUpgrade started
+          else
+            ./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs namenode
+          fi
         resource-set: name-resources
         readiness-check:
           cmd: |
             set -e
             export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/)
+            export service_state=$(./{{TASKCFG_ALL_HDFS_VERSION}}/bin/hdfs haadmin -getServiceState $TASK_NAME | grep -i 'standby')
+            export LATEST_FSIMAGE_VERSION=-65
+            export CURRENT_FSIMAGE_VERSION=$(cat name-data/current/VERSION | grep "layoutVersion" | cut -d'=' -f2)
```

Comment:
you don't need to cat a file into grep, you can more simply
```
grep "layoutVersion" name-data/current/VERSION | cut -d....
```

Reply:
Done